tests/unit/SpecHelper.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+const iframes = [];
+
+/**
+ * Create a doc for unit tests.
+ * @param {string[]} plugins Paths of plugins to load and run. Example: `/src/core/algorithms.js`
+ * @param {object} config JSON-serializable respecConfig object.
+ * @return {Promise<Document>}
+ */
+export function makePluginDoc(
+  plugins,
+  config = {},
+  { head = `<meta charset="UTF-8" />`, body = "" } = {}
+) {
+  const html = `
+    <!DOCTYPE html>
+    <html lang="en">
+      <head>
+        ${head}
+        <script>
+          var respecConfig = ${JSON.stringify(config || {}, null, 2)};
+        </script>
+        <script type="module">
+          const plugins = ${JSON.stringify(plugins)};
+          const allPlugins = ["/src/core/base-runner.js"]
+            .concat(plugins)
+            .map(p => "/base" + p);
+          Promise.all(allPlugins.map(plug => import(plug)))
+            .then(([baseRunner, ...importedPlugins]) =>
+              baseRunner.runAll(importedPlugins)
+            )
+            .catch(err => {
+              console.error(err);
+              if (document.respec) {
+                document.respec.errors.push(err);
+              } else {
+                Object.defineProperty(document, "respec", {
+                  value: { ready: Promise.reject(err) },
+                });
+              }
+            });
+        </script>
+      </head>
+      <body>${body}</body>
+    </html>
+  `;
+
+  return new Promise((resolve, reject) => {
+    const ifr = document.createElement("iframe");
+    const timeoutId = setTimeout(() => {
+      reject(new Error(`Timed out waiting.`));
+    }, jasmine.DEFAULT_TIMEOUT_INTERVAL);
+    ifr.addEventListener("load", async () => {
```

Comment:
It might be a good opportunity to refactor out the async function here to make this whole block a little smaller. Maybe curry it to take the `timeoutId`.